### ISSUE TITLE
Salt the hash on per user basis

### ISSUE THREAD

**Issue by ChaosEngine** [the-urlist/BlazorSWA#19]
[Here](https://github.com/the-urlist/BlazorSWA/blob/194aedfe3dde5331d0484f5527d2e056453264ff/Api/SaveLinks.cs#L66) You guys are doing hashing of provider+user combo but...
...for real unbreakable combi I would add generated salt and use pattern like `$"{provider}_{user}_"salt}"` per user

**Comment by burkeholland** [comment]
Is that because the salt then isn't a single value? How is it more secure if the attacker can still get access to the SALT via the DB?

Thanks for this suggestion! We're def doing this provided it's more secure.

**Comment by ChaosEngine** [comment]
> Is that because the salt then isn't a single value? How is it more secure if the attacker can still get access to the SALT via the DB?
> 
Yes. If attacker would create it's own account and get username (from leak?) then it would be infinitely ore easy to factor out algorithm and reverting all other users from it's hash without [per-user salt](https://en.wikipedia.org/wiki/Salt_(cryptography)).

> Thanks for this suggestion! We're def doing this provided it's more secure.
